src/main/java/org/graylog/plugins/threatintel/tools/PrivateNet.java

```diff
@@ -17,26 +17,25 @@
 package org.graylog.plugins.threatintel.tools;
 
 import com.google.common.net.InetAddresses;
-import org.apache.commons.net.util.SubnetUtils;
 
-public class PrivateNet {
+import java.net.Inet6Address;
+import java.net.InetAddress;
 
-    public static final SubnetUtils.SubnetInfo TEN = new SubnetUtils("10.0.0.0/8").getInfo();
-    public static final SubnetUtils.SubnetInfo ONE_HUNDRED_SEVENTY_TWO = new SubnetUtils("172.16.0.0/12").getInfo();
-    public static final SubnetUtils.SubnetInfo ONE_HUNDRED_NINETY_TWO = new SubnetUtils("192.168.0.0/16").getInfo();
+public class PrivateNet {
 
-    /**
-     * Checks if an IPv4 address is part of a private network as defined in RFC 1918.
+   /**
+     * Checks if an IPv4 address is part of a private network as defined in RFC 1918. This ignores IPv6 addresses for now and always returns false for them.
      *
      * @param ip The IPv4 address to check
      * @return
      */
     public static boolean isInPrivateAddressSpace(String ip) {
-        if(!InetAddresses.isInetAddress(ip)) {
+        InetAddress inetAddress = InetAddresses.forString(ip);
+        if (inetAddress instanceof Inet6Address) {
+            // we don't deal with IPv6 unique local addresses currently.
             return false;
```

Comment:
How about we return a boxed `Boolean` here, and keep returning `null` for v6 addresses?

Reply:
`null` could be an option, but I'd like to avoid doing it if possible.
An alternative is to use https://www.rfc-editor.org/rfc/rfc4193.txt ranges (`FC00::/7`), which I think is technically correct, but probably rarely used in practice.

Reply:
that's what `isSiteLocalAddress` would be doing for v6. But I'm wondering if it wouldn't be more practical to also include link local addresses `fe80::/10` as well.

Reply:
I think link-local addresses are out of scope for this function: the original version also didn't consider link-local IPv4 address, because they shouldn't be routed anyway.

For the purposes of this function, I'll now update the PR to include the unique local addresses in IPv6 and properly return `true` for those.